### PR TITLE
big_mod_exp syscall: exclude inputs of numbers larger than 4096-bits

### PR DESCRIPTION
**Pull request by valiksinev**
#### Problem
The changes related to Big integer modular exponentiation (EIP-198). The calculation is implemented as syscall.

The formula that was used to approximate the compute budgets for large number exponentiation seems to under-approximate the costs for exponentiation of very large numbers (i.e. >>1000 byte numbers). This could lead to a negative impact on the network if the syscall was to be invoked with very large numbers.
In practice, it seems like any normal/non-adversarial use of the syscall will be on numbers of size less than or equal to 4096-bits (i.e. for RSA-4096)

#### Summary of Changes
Inside syscall all inputs are limited by value 4069 bits (512 bytes).  
If the length is exceeded, an error SyscallError::InvalidLength returned.

### CODE REVIEW

**Comment by t-nelson** [comment]
thanks!  mind adding some test cases to cover this new logic?

File: programs/bpf_loader/src/syscalls/mod.rs [hunk 1]
```diff
@@ -1720,6 +1720,10 @@ declare_syscall!(
         .get(0)
         .ok_or(SyscallError::InvalidLength)?;
 
+        if params.base_len > 512 || params.exponent_len > 512 || params.modulus_len > 512 {
```
[the remark this answers is not in the document]
Comment (samkim-crypto) [reply programs/bpf_loader/src/syscalls/mod.rs:1723]: This syscall itself is feature gated by https://github.com/solana-labs/solana/issues/29882, which is not activated yet and should not be activated without this change. So does this change need to be gated by a different feature gate?

It seems like this change should either be absorbed into the original feature gate or the original feature gate should be killed and replaced by a new feature gate that includes this change. Did you mean the latter here?

[the remark this answers is not in the document]
Comment (mvines) [reply programs/bpf_loader/src/syscalls/mod.rs:1723]: oh, great! you're totally right then, no need for a feature gate then

**Comment by mvines** [comment]
I've moved https://github.com/solana-labs/solana/issues/29882 off the activation schedule, and put it into https://github.com/solana-labs/solana/wiki/Feature-Gate-Activation-Schedule#features-are-blocked as any activation of that feature cannot happen until this PR is merged and backported to v1.16

**Comment by derrekr** [comment]
Ping @valiksinev do you have a test case yet?

**Comment by mvines** [comment]
The changes in this PR were merged via https://github.com/solana-labs/solana/pull/32520

**Pull request closed without merging**